nautobot/core/filters.py

```diff
@@ -462,6 +445,21 @@ class SearchFilter(MappedPredicatesFilterMixin, django_filters.CharFilter):
     label = "Search"
 
 
+class TagFilter(NaturalKeyOrPKMultipleChoiceFilter):
+    """
+    Match on one or more assigned tags. If multiple tags are specified (e.g. ?tag=foo&tag=bar), the queryset is filtered
+    to objects matching all tags.
+    """
+
+    def __init__(self, *args, **kwargs):
+        from nautobot.extras import models as extras_models  # avoid circular import
+
```

Comment:
Might need to still `kwargs.setdefault("field_name", "tags")` to keep this working for any apps that are still using the old `tag = TagFilter()` pattern.

Reply:
```suggestion

        kwargs.setdefault("field_name", "tags")
```